modules/database/test/std/rec/aiTest.c

```diff
@@ -0,0 +1,430 @@
+
+    // number of tests = 29
+}
+
+static void test_aftc(void){
+    const double h = 5000;
+    const double hh = 7500;
+    const double l = 200;
+    const double ll = -20;
+    const double aftc = 3;
+    testMonitor* test_mon = NULL;
+    epicsTimeStamp startTime;
+    epicsTimeStamp endTime;
+    double diffTime = 0;
+
+    /* set soft channel */
+    testdbPutFieldOk("test_ai_rec3.DTYP", DBF_STRING, "Raw Soft Channel");
+    testdbPutFieldOk("test_ai_rec3.INP", DBF_STRING, "test_ai_link_rec3.VAL");
+    testdbPutFieldOk("test_ai_link_rec3.FLNK", DBF_STRING, "test_ai_rec3");
+
+    /* set alarm parameters */
+    testdbPutFieldOk("test_ai_rec3.HIGH", DBF_DOUBLE, h);
+    testdbPutFieldOk("test_ai_rec3.HIHI", DBF_DOUBLE, hh);
+    testdbPutFieldOk("test_ai_rec3.LOW", DBF_DOUBLE, l);
+    testdbPutFieldOk("test_ai_rec3.LOLO", DBF_DOUBLE, ll);
+    testdbPutFieldOk("test_ai_rec3.HHSV", DBF_SHORT, menuAlarmSevrMAJOR);
+    testdbPutFieldOk("test_ai_rec3.HSV", DBF_SHORT, menuAlarmSevrMINOR);
+    testdbPutFieldOk("test_ai_rec3.LSV", DBF_SHORT, menuAlarmSevrMINOR);
+    testdbPutFieldOk("test_ai_rec3.LLSV", DBF_SHORT, menuAlarmSevrMAJOR);
+
+    /* test AFTC using a monitor and time stamps */
+    testdbPutFieldOk("test_ai_rec3.SCAN", DBF_SHORT, menuScan_1_second);
+    testdbPutFieldOk("test_ai_rec3.AFTC", DBF_DOUBLE, aftc);
+
+    /* set HIHI alarm VAL */
+    testdbPutFieldOk("test_ai_link_rec3.VAL", DBF_DOUBLE, 19998.0);
+
+    /* Create test monitor for alarm SEVR */
+    test_mon = testMonitorCreate("test_ai_rec3.VAL", DBE_ALARM, 0);
+
+    /* Get start time */
+    epicsTimeGetCurrent(&startTime);
+
+    /* wait for monitor to trigger on the new alarm status*/
+    testMonitorWait(test_mon);
+    epicsTimeGetCurrent(&endTime);
+
+    /* Verify that alarm status is now MAJOR */
+    testdbGetFieldEqual("test_ai_rec3.SEVR", DBF_SHORT, menuAlarmSevrMAJOR);
+    
+    /* Verify that time is at least equal to configured aftc */
+    diffTime = epicsTimeDiffInSeconds(&endTime, &startTime);
+    testOk(diffTime >= aftc, "ATFC time %lf", diffTime);
+ 
+    // number of tests = 16
+}
+
+MAIN(aiTest) {
+
+#ifdef _WIN32
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || \
+    (defined(_MINGW) && defined(_TWO_DIGIT_EXPONENT))
+    _set_output_format(_TWO_DIGIT_EXPONENT);
+#endif
+#endif
+
+    testPlan(6+6+11+10+12+14+18+15+6+29);
+
+    testdbPrepare();   
+    testdbReadDatabase("recTestIoc.dbd", NULL, NULL);
+    recTestIoc_registerRecordDeviceDriver(pdbbase);
+
+    testdbReadDatabase("aiTest.db", NULL, NULL);
+    
+    eltc(0);
+    testIocInitOk();
+    eltc(1);
+
+    test_soft_input();
+    test_raw_soft_input();
+    test_operator_display();
+    test_no_linr_unit_conversion();
+    test_slope_linr_unit_conversion();
+    test_linear_linr_unit_conversion();
+    test_bpt_conversion();
+    test_smoothing_filter();
+    test_udf();
+    test_alarm();
+    /* TODO: Investigate and make the AFTC test work correctly 
```

Comment:
If these tests aren't working, it might be a good idea to document better what the issue is.